storage/remote/queue_manager.go

```diff
@@ -844,38 +846,70 @@ func (s *shards) sendSamplesWithBackoff(ctx context.Context, samples []prompb.Ti
 		return err
 	}
 
+	try := 0
+
 	for {
 		select {
 		case <-ctx.Done():
 			return ctx.Err()
 		default:
 		}
-		begin := time.Now()
-		err := s.qm.client().Store(ctx, req)
-
-		s.qm.metrics.sentBatchDuration.Observe(time.Since(begin).Seconds())
 
+		err = s.attemptSendSamples(ctx, samples, buf, highest, try)
```

Comment:
Let's pass down `req` here rather than `buf`. `buf` is just for allocation optimizations and it is nice to keep its scope as limited as possible.

Reply:
:+1: Sounds good, I was 50/50 on that one.